lib/mayaUsd/ufe/UsdUndoSetKindCommand.h

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright 2021 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+#ifndef MAYAUSD_UFE_USD_UNDO_SET_KIND_COMMAND_H
+#define MAYAUSD_UFE_USD_UNDO_SET_KIND_COMMAND_H
+
+#include <mayaUsd/base/api.h>
+#include <mayaUsd/undo/UsdUndoableItem.h>
+
+#include <pxr/base/tf/token.h>
+#include <pxr/usd/usd/prim.h>
+
+#include <ufe/undoableCommand.h>
+
+#include <memory>
+
+namespace MAYAUSD_NS_DEF {
+namespace ufe {
+
+//! Undoable command for setting the kind metadata of a UsdPrim.
+class MAYAUSD_CORE_PUBLIC UsdUndoSetKindCommand : public Ufe::UndoableCommand
+{
+public:
+    using Ptr = std::shared_ptr<UsdUndoSetKindCommand>;
+
+    UsdUndoSetKindCommand(const PXR_NS::UsdPrim& prim, const PXR_NS::TfToken& kind);
+    ~UsdUndoSetKindCommand() override = default;
+
+    UsdUndoSetKindCommand(const UsdUndoSetKindCommand&) = delete;
+    UsdUndoSetKindCommand& operator=(const UsdUndoSetKindCommand&) = delete;
+    UsdUndoSetKindCommand(UsdUndoSetKindCommand&&) = delete;
+    UsdUndoSetKindCommand& operator=(UsdUndoSetKindCommand&&) = delete;
```

Comment:
I see we copy this pattern around making the declaration of this trivial class busier than it has to be.

The base class `Ufe::UndoableCommand` already explicitly deletes the copy constructor. This will make the compiler implicitly delete any derived class copy constructors. So line 41 is not needed.
Lack of a copy constructor will also make the compiler implicitly delete the move constructor. So line 43 is not needed.

Finally, we should do a better job in the base class and delete the copy assignment operator, this would make the compiler implicitly delete the copy assignment operator and move the assignment operator from the derived class. This is not currently done, so lines 42 and 44 have to stay.

Reply:
That's a good point. I really don't have a strong opinion here. I personally am not a huge fan of implicit philosophy whether it is C++ or any other languages.